### PR TITLE
Removed some mystery from creating endpoint custom code

### PR DESCRIPTION
**Pull request by bamandel**
I've created a few new classes to allow people to more easily code for custom endpoints. The goal here is to help people write response code that will meet the expectations of the fields implementation in JS.

### CODE REVIEW

File: alert-common/src/main/java/com/synopsys/integration/alert/common/descriptor/config/field/endpoint/EndpointResponse.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+import org.springframework.http.ResponseEntity;
+
+import com.synopsys.integration.alert.common.action.CustomEndpointManager;
+import com.synopsys.integration.alert.common.exception.AlertException;
+import com.synopsys.integration.alert.common.rest.model.FieldValueModel;
+
+public abstract class EndpointResponse<R> {
+
+    public EndpointResponse(final String fieldKey, final CustomEndpointManager customEndpointManager) throws AlertException {
```
Comment (gkillough) [inline alert-common/src/main/java/com/synopsys/integration/alert/common/descriptor/config/field/endpoint/EndpointResponse.java]: This object isn't a response, so the name is confusing.

Comment (bamandel) [reply alert-common/src/main/java/com/synopsys/integration/alert/common/descriptor/config/field/endpoint/EndpointResponse.java]: This object is part of the response and helps build it. Do you recommend another name?

File: alert-common/src/main/java/com/synopsys/integration/alert/common/descriptor/config/ui/ProviderDistributionCustomEndpoint.java [hunk 2]
```diff
@@ -32,44 +33,44 @@
...
+import com.synopsys.integration.alert.common.descriptor.config.field.endpoint.table.TableSelectEndpointResponse;
 import com.synopsys.integration.alert.common.exception.AlertException;
 import com.synopsys.integration.alert.common.persistence.accessor.ProviderDataAccessor;
-import com.synopsys.integration.alert.common.persistence.model.ProviderProject;
 import com.synopsys.integration.alert.common.rest.ResponseFactory;
 import com.synopsys.integration.alert.common.rest.model.FieldValueModel;
 
 @Component
-public class ProviderDistributionCustomEndpoint {
+public class ProviderDistributionCustomEndpoint extends TableSelectEndpointResponse {
```
Comment (jamesrichard91) [inline alert-common/src/main/java/com/synopsys/integration/alert/common/descriptor/config/ui/ProviderDistributionCustomEndpoint.java]: Is this an endpoint or a response object? The names are confusing

File: src/main/java/com/synopsys/integration/alert/channel/jira/cloud/web/JiraCustomEndpoint.java [hunk 3]
```diff
@@ -51,7 +53,7 @@
 import com.synopsys.integration.rest.request.Response;
 
 @Component
-public class JiraCustomEndpoint {
+public class JiraCustomEndpoint extends ButtonEndpointResponse {
```
Comment (jamesrichard91) [inline src/main/java/com/synopsys/integration/alert/channel/jira/cloud/web/JiraCustomEndpoint.java]: Is this an endpoint or a response object? The names are confusing

File: alert-common/src/main/java/com/synopsys/integration/alert/common/descriptor/config/ui/ChannelDistributionCustomEndpoint.java [hunk 4]
```diff
@@ -44,40 +44,46 @@
...
 
     @Autowired
     public ChannelDistributionCustomEndpoint(CustomEndpointManager customEndpointManager, DescriptorMap descriptorMap, ResponseFactory responseFactory, Gson gson) throws AlertException {
         this.descriptorMap = descriptorMap;
-        this.responseFactory = responseFactory;
-        this.gson = gson;
 
-        customEndpointManager.registerFunction(ChannelDistributionUIConfig.KEY_PROVIDER_NAME, this::retrieveProviders);
-        customEndpointManager.registerFunction(ChannelDistributionUIConfig.KEY_CHANNEL_NAME, this::retrieveChannels);
+        new ProvidersFunction(customEndpointManager, responseFactory, gson);
```
Comment (gkillough) [inline alert-common/src/main/java/com/synopsys/integration/alert/common/descriptor/config/ui/ChannelDistributionCustomEndpoint.java]: These are unassigned.

**Pull request merged**